module/users/helper.js

```diff
@@ -1119,6 +1142,120 @@ module.exports = class UsersHelper {
       }
     });
   }
+    /**
+   * Get overall Stats of the User
+   * @method
+   * @name getAllStats
+   * @param {String} userToken - userToken
+   * @param {String} type - type
+   * @returns {Object} - returns overall stats or targeted records based on type
+   */
+  static getAllStats({ userToken, type }) {
+    
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (type) {
+          switch (type) {
+            case "observation":
+              let observationInfo = await surveyService.getObservationInfo({
+                userToken,
+              });
+
+              resolve({
+                type: type,
+                data: observationInfo.data,
+              });
+
+              break;
+            case "project":
+              let projectsStartedByUserAPICall =
+                improvementProjectService.listProjectOverviewInfo({
+                  stats: "false",
+                  userToken,
+                  userInvolvement:'creator'
+                });
+
+              let projectsConsumedByUserAPICall =
+                improvementProjectService.listProjectOverviewInfo({
+                  stats: "false",
+                  userToken,
+                  userInvolvement:'consumed'
+                });
+
+              Promise.all([
+                projectsStartedByUserAPICall,
+                projectsConsumedByUserAPICall
+              ])
+                .then((responses) => {
+                  const [response1, response2] =
+                    responses;
+
+                  resolve({
+                    type,
+                    data:{
+                      projectsStartedByUser: response1.data,
+                      projectsConsumedByUser: response2.data,
+                    }
+                  });
+                })
+                .catch((error) => {
+                  
+                  throw new Error("Something went wrong...");
+                });
+
+              break;
+              default:
+                throw new Error('Invalid Type passed.')
+          }
+        } else {
+          let projectsStartedByUserStatsAPICall =
+            improvementProjectService.listProjectOverviewInfo({
+              stats: "true",
+              userToken,
```

Comment:
include survey as well @borkarsaish65

Reply:
survey part is handled by Praveen

Reply:
i agree but there will be only one controller which will be responsible for giving response @borkarsaish65

Reply:
okay once my side of work is done let praveen work on the same branch to add his changes

Reply:
I have added the survey and program code